File: Algorithm_434_Number of Segments in a String.cpp
#include <sstream>
using namespace std;
class Solution {
public:
    int countSegments(string s) {
        s=' '+s;
        int count=0;
        for (int i=1;i<s.size();i++) if (s[i-1]==' ' && s[i]!=' ') count++;
        return count;
    }
};

File: Algorithm_434_Number of Segments in a String.java
// 先split
// 不能直接返回split后的数组的长度，因为题目要求如果segment是空字符串""，则不算

class Solution {
    public int countSegments(String s) {
        int count = 0;
        String[] segments = s.split(" ");
        for (String seg : segments) {
            if (!"".equals(seg)) {
                count++;
            }
        }
        return count;
    }
}